### DISCUSSION TITLE
How to access `@children`'s size?

### DISCUSSION THREAD

**Discussion by Enyium** [slint-ui/slint#6254] [Q&A] [answered]
Consider [this](https://slintpad.com/?snippet=import+%7B+Button+%7D+from+%22std-widgets.slint%22%3B%0A%0Acomponent+Container+inherits+Rectangle+%7B%0A++++background%3A+pink%3B%0A%0A++++Rectangle+%7B%0A++++++++width%3A+100px%3B+%2F%2F+How+to+use+%60%40children%60%27s+size%3F%0A++++++++height%3A+100px%3B%0A++++++++background%3A+lime%3B%0A%0A++++++++%40children%0A++++%7D%0A%7D%0A%0Aexport+component+Demo+%7B%0A++++width%3A+300px%3B%0A++++height%3A+300px%3B%0A%0A++++Container+%7B%0A++++++++Button+%7B+text%3A+%22Foo%22%3B+%7D%0A++++%7D%0A%7D%0A&style=native) code:

```slint
import { Button } from "std-widgets.slint";

component Container inherits Rectangle {
    background: pink;

    Rectangle {
        width: 100px; // How to use `@children`'s size?
        height: 100px;
        background: lime;

        @children
    }
}

export component Demo {
    width: 300px;
    height: 300px;

    Container {
        Button { text: "Foo"; }
    }
}
```

How can I bind to `@children`'s size, so that the `lime`-colored surface is as large as them?

**Comment by felipetesc** [comment]
The thing is, do you actually need @children's size to make as large as them ?

```slint
import { Button } from "std-widgets.slint";

component Container inherits Rectangle {
    background: pink;
    in-out property inner-width<length> : outerRect.width;
    n-out property inner-height<length> : outerRect.height;
    
     outerRect := Rectangle {
        width: 100px; // How to use `@children`'s size?
        height: 100px;
        background: lime;
        innerRect := Rectangle{
                  width: root.inner-width;
                  height:  root.height;
                  background: parent.background;
                  @children
        }
        
    }
}

export component Demo {
    width: 300px;
    height: 300px;

    Container {
        inner-width: bt.width;
        inner-height: bt.height;
        bt := Button { text: "Foo"; }
    }
}
```


or something like that ....

**Comment by ogoffart** [comment]
i'd use a layout:
```slint
Rectangle {
        width: l.preferred-width;
        height: l.preferred-height;
        background: lime;
        l := VerticalLayout {  @children }
}
```

**Reply by Enyium** [reply, marked as answer]
I think, looking forward, a `CellLayout` would be the go-to solution for this. See <https://github.com/slint-ui/slint/issues/4800#issuecomment-2368661375>.
